share/getters/shrex.go

```diff
@@ -33,7 +33,7 @@ func (sg *ShrexGetter) GetShare(ctx context.Context, root *share.Root, row, col
 }
 
 func (sg *ShrexGetter) GetEDS(ctx context.Context, root *share.Root) (*rsmt2d.ExtendedDataSquare, error) {
-	for {
+	for ctx.Err() != nil {
```

Comment:
huh? shouldn't it be the other way around?  and even so, can we stick w wrapping w select pattern pls?

Reply:
yes